### ISSUE TITLE
1.4.13 Live TV playback fails

### ISSUE THREAD

**Issue by jcdick1** [jellyfin-archive/jellyfin-roku-legacy#694]
**Describe the bug**
Any live TV selection results in "An error was encountered while playing this item."

**To Reproduce**

1. Go to 'Live TV'
2. Select any program
3. Select "View Channel"
4. Get error.

**Expected behavior**
Should be able to view the stream

**Logs**
```
[2022-07-15 10:27:36.993 -05:00] [INF] User policy for "user". EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
[2022-07-15 10:27:36.993 -05:00] [INF] StreamBuilder.BuildVideoItem( Profile="Anonymous Profile", Path="http://192.168.0.178", AudioStreamIndex=null, SubtitleStreamIndex=null ) => ( PlayMethod=Transcode, TranscodeReason=ContainerNotSupported ) "media:/videos/82b41a89-9cfc-72a7-3b66-d16e09454788/master.m3u8?MediaSourceId=native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55&VideoCodec=h264,mpeg2video,h265,hevc,mpeg2video&AudioCodec=AC3&AudioStreamIndex=-1&VideoBitrate=139552000&AudioBitrate=448000&api_key=<token>&TranscodingMaxAudioChannels=6&RequireAvc=false&SegmentContainer=ts&MinSegments=1&BreakOnNonKeyFrames=True&h264-profile=high,main,baseline,constrainedbaseline&h264-level=51&aac-audiochannels=6&TranscodeReasons=ContainerNotSupported"
[2022-07-15 10:27:36.995 -05:00] [INF] Opening channel stream from "Emby", external channel Id: "hdhr_845"
[2022-07-15 10:27:36.995 -05:00] [INF] Streaming Channel "hdhr_845"
[2022-07-15 10:27:36.995 -05:00] [INF] GetChannelStream: channel id: "hdhr_845". stream id: "native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55" profile: "native"
[2022-07-15 10:27:36.996 -05:00] [INF] Opening "SharedHttpStream" Live stream from "http://192.168.0.178:5004/auto/v845"
[2022-07-15 10:27:37.548 -05:00] [INF] Beginning "SharedHttpStream" stream to "/dev/shm/873a81d22ec84e10936b05076d2062a9.ts"
[2022-07-15 10:27:37.549 -05:00] [INF] Live stream opened after 553.2973ms
[2022-07-15 10:27:37.549 -05:00] [INF] Returning mediasource streamId "native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55", mediaSource.Id "native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55", mediaSource.LiveStreamId null
[2022-07-15 10:27:37.552 -05:00] [INF] Live tv media info probe took "0.0022548" seconds
[2022-07-15 10:27:37.553 -05:00] [INF] Live stream opened: MediaSourceInfo { Protocol: Http, Id: "native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55", Path: "http://192.168.0.25:8096/LiveTv/LiveStreamFiles/873a81d22ec84e10936b05076d2062a9/stream.ts", EncoderPath: null, EncoderProtocol: null, Type: Default, Container: "mpegts", Size: null, Name: null, IsRemote: False, ETag: null, RunTimeTicks: null, ReadAtNativeFramerate: False, IgnoreDts: True, IgnoreIndex: False, GenPtsInput: False, SupportsTranscoding: True, SupportsDirectStream: True, SupportsDirectPlay: False, IsInfiniteStream: True, RequiresOpening: True, OpenToken: null, RequiresClosing: True, LiveStreamId: "a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55", BufferMs: 0, RequiresLooping: False, SupportsProbing: True, VideoType: null, IsoType: null, Video3DFormat: null, MediaStreams: [MediaStream { Codec: "mpeg2video", CodecTag: null, Language: null, ColorRange: "tv", ColorSpace: null, ColorTransfer: null, ColorPrimaries: null, DvVersionMajor: null, DvVersionMinor: null, DvProfile: null, DvLevel: null, RpuPresentFlag: null, ElPresentFlag: null, BlPresentFlag: null, DvBlSignalCompatibilityId: null, Comment: null, TimeBase: "1/90000", CodecTimeBase: null, Title: null, VideoRange: "SDR", VideoRangeType: "SDR", VideoDoViTitle: null, LocalizedUndefined: null, LocalizedDefault: null, LocalizedForced: null, LocalizedExternal: null, DisplayTitle: "1080i MPEG2VIDEO SDR", NalLengthSize: null, IsInterlaced: True, IsAVC: null, ChannelLayout: null, BitRate: 20000000, BitDepth: 8, RefFrames: 1, PacketLength: null, Channels: null, SampleRate: null, IsDefault: False, IsForced: False, Height: 1080, Width: 1920, AverageFrameRate: 29.97003, RealFrameRate: 29.97003, Profile: "Main", Type: Video, AspectRatio: "16:9", Index: -1, Score: null, IsExternal: False, DeliveryMethod: null, DeliveryUrl: null, IsExternalUrl: null, IsTextSubtitleStream: False, SupportsExternalStream: False, Path: null, PixelFormat: "yuv420p", Level: 4, IsAnamorphic: False }, MediaStream { Codec: "ac3", CodecTag: "AC-3", Language: null, ColorRange: null, ColorSpace: null, ColorTransfer: null, ColorPrimaries: null, DvVersionMajor: null, DvVersionMinor: null, DvProfile: null, DvLevel: null, RpuPresentFlag: null, ElPresentFlag: null, BlPresentFlag: null, DvBlSignalCompatibilityId: null, Comment: null, TimeBase: "1/90000", CodecTimeBase: null, Title: null, VideoRange: null, VideoRangeType: null, VideoDoViTitle: null, LocalizedUndefined: null, LocalizedDefault: null, LocalizedForced: null, LocalizedExternal: null, DisplayTitle: "Dolby Digital - 5.1", NalLengthSize: null, IsInterlaced: False, IsAVC: null, ChannelLayout: "5.1", BitRate: 384000, BitDepth: null, RefFrames: null, PacketLength: null, Channels: 6, SampleRate: 48000, IsDefault: False, IsForced: False, Height: null, Width: null, AverageFrameRate: null, RealFrameRate: null, Profile: null, Type: Audio, AspectRatio: null, Index: -1, Score: null, IsExternal: False, DeliveryMethod: null, DeliveryUrl: null, IsExternalUrl: null, IsTextSubtitleStream: False, SupportsExternalStream: False, Path: null, PixelFormat: null, Level: 0, IsAnamorphic: null }], MediaAttachments: [], Formats: [], Bitrate: 20384000, Timestamp: null, RequiredHttpHeaders: [], TranscodingUrl: null, TranscodingSubProtocol: null, TranscodingContainer: null, AnalyzeDurationMs: 3000, TranscodeReasons: 0, DefaultAudioStreamIndex: null, DefaultSubtitleStreamIndex: null, VideoStream: MediaStream { Codec: "mpeg2video", CodecTag: null, Language: null, ColorRange: "tv", ColorSpace: null, ColorTransfer: null, ColorPrimaries: null, DvVersionMajor: null, DvVersionMinor: null, DvProfile: null, DvLevel: null, RpuPresentFlag: null, ElPresentFlag: null, BlPresentFlag: null, DvBlSignalCompatibilityId: null, Comment: null, TimeBase: "1/90000", CodecTimeBase: null, Title: null, VideoRange: "SDR", VideoRangeType: "SDR", VideoDoViTitle: null, LocalizedUndefined: null, LocalizedDefault: null, LocalizedForced: null, LocalizedExternal: null, DisplayTitle: "1080i MPEG2VIDEO SDR", NalLengthSize: null, IsInterlaced: True, IsAVC: null, ChannelLayout: null, BitRate: 20000000, BitDepth: 8, RefFrames: 1, PacketLength: null, Channels: null, SampleRate: null, IsDefault: False, IsForced: False, Height: 1080, Width: 1920, AverageFrameRate: 29.97003, RealFrameRate: 29.97003, Profile: "Main", Type: Video, AspectRatio: "16:9", Index: -1, Score: null, IsExternal: False, DeliveryMethod: null, DeliveryUrl: null, IsExternalUrl: null, IsTextSubtitleStream: False, SupportsExternalStream: False, Path: null, PixelFormat: "yuv420p", Level: 4, IsAnamorphic: False } }
[2022-07-15 10:27:37.556 -05:00] [INF] User policy for "user". EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
[2022-07-15 10:27:37.556 -05:00] [INF] StreamBuilder.BuildVideoItem( Profile="Anonymous Profile", Path="http://192.168.0.25:8096/LiveTv/LiveStreamFiles/873a81d22ec84e10936b05076d2062a9/stream.ts", AudioStreamIndex=1, SubtitleStreamIndex=-1 ) => ( PlayMethod=Transcode, TranscodeReason=ContainerNotSupported ) "media:/videos/82b41a89-9cfc-72a7-3b66-d16e09454788/master.m3u8?MediaSourceId=native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55&VideoCodec=h264,mpeg2video,h265,hevc,mpeg2video&AudioCodec=ac3&AudioStreamIndex=-1&VideoBitrate=139616000&AudioBitrate=384000&AudioSampleRate=48000&MaxFramerate=29.97003&api_key=<token>&LiveStreamId=a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55&SubtitleMethod=Encode&TranscodingMaxAudioChannels=6&RequireAvc=false&SegmentContainer=ts&MinSegments=1&BreakOnNonKeyFrames=True&mpeg2video-level=4&mpeg2video-videobitdepth=8&mpeg2video-profile=main&mpeg2video-audiochannels=6&h264-profile=high,main,baseline,constrainedbaseline&h264-level=51&aac-audiochannels=6&TranscodeReasons=ContainerNotSupported"
[2022-07-15 10:27:37.558 -05:00] [WRN] Slow HTTP Response from "https://jf.cadmed.net/Items/82b41a899cfc72a73b66d16e09454788/PlaybackInfo?audiostreamindex=1&AutoOpenLiveStream=true&IsPlayback=true&MaxStaticBitrate=140000000&MaxStreamingBitrate=140000000&StartTimeTicks=0&SubtitleStreamIndex=-1&UserId=ac78931532bc477887254b04c072f977" to "192.168.0.116" in 0:00:00.5700978 with Status Code 200
[2022-07-15 10:27:37.817 -05:00] [INF] Playback stopped reported by app "Jellyfin Roku" "1.4.12" playing "HD SYFY BB". Stopped at "0" ms
[2022-07-15 10:27:37.846 -05:00] [INF] Live stream "native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55" consumer count is now 0
[2022-07-15 10:27:37.846 -05:00] [INF] Closing live stream "a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55"
[2022-07-15 10:27:37.847 -05:00] [INF] Closing "SharedHttpStream"
[2022-07-15 10:27:37.847 -05:00] [INF] Live stream "a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_native_f9f9b791a1db9f37d072754a4c7d85dc_a57953ea4242097a90dcd3024b93ba55" closed successfully
[2022-07-15 10:27:37.848 -05:00] [INF] Copying of "SharedHttpStream" to "/dev/shm/873a81d22ec84e10936b05076d2062a9.ts" was canceled
[2022-07-15 10:27:37.849 -05:00] [INF] Deleting temp file "/dev/shm/873a81d22ec84e10936b05076d2062a9.ts"
[2022-07-15 10:27:37.855 -05:00] [INF] Playback stop did not have a tracker : "3303cb83-cdc5-5928-bea3-070c8d7d0909-ac78931532bc477887254b04c072f977-82b41a899cfc72a73b66d16e09454788"
```

**Comment by funkybunch** [comment]
I suspect this is a duplicate of #670.  Can you open playback settings (*)->Settings->Playback and enable MPEG-2 direct streaming?

That may not work on server version 10.8.1, but it's a known issue that with server version 10.8+ the Jellyfin Roku app is not transcoding properly.

**Comment by funkybunch** [comment]
PR #681 should resolve this once a release is cut with that code.

**Comment by whiteowl3** [comment]
This fix has been released. Please consider testing the new version and updating your issue accordingly @jcdick1

**Comment by funkybunch** [comment]
Updated to the build 14 hot fix and that has resolved the transcoding issue for me.

**Comment by cewert** [comment]
Fixed by #699
